### PR TITLE
wolfssl: update to 5.6.3

### PR DESCRIPTION
**Pull request by PolynomialDivision**
Release Notes:
https://github.com/wolfSSL/wolfssl/releases/tag/v5.6.0-stable

Refresh patch:
- 100-disable-hardening-check.patch

### CODE REVIEW

**Comment by hauke** [comment]
It looks like the API changed:
````
 make[3] -C package/utils/px5g-wolfssl compile
    ERROR: package/utils/px5g-wolfssl failed to build (build variant: wolfssl).
make -r world: build failed. Please re-run make with -j1 V=s or V=sc for a higher verbosity level to see what's going on
make: *** [/__w/openwrt/openwrt/openwrt/include/toplevel.mk:231: world] Error 1
====== Make errors from logs/package/utils/px5g-wolfssl/wolfssl/compile.txt ======
In file included from /__w/openwrt/openwrt/openwrt/staging_dir/target-mips-openwrt-linux-musl_musl/usr/include/wolfssl/wolfcrypt/asn.h:46,
                 from px5g-wolfssl.c:11:
/__w/openwrt/openwrt/openwrt/staging_dir/target-mips-openwrt-linux-musl_musl/usr/include/wolfssl/wolfcrypt/integer.h:37:14: fatal error: wolfssl/wolfcrypt/sp_int.h: No such file or directory
   37 |     #include <wolfssl/wolfcrypt/sp_int.h>
      |              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[3]: *** [Makefile:53: /__w/openwrt/openwrt/openwrt/build_dir/target-mips-openwrt-linux-musl_musl/px5g-wolfssl-wolfssl/px5g-wolfssl/.built] Error 1
time: package/utils/px5g-wolfssl/wolfssl/compile#0.17#0.05#0.23
Error: Process completed with exit code 2.
````

**Comment by PolynomialDivision** [comment]
On mt7622 it builds completely fine. I will try now x86/64 build.

**Comment by hauke** [comment]
I can not reproduce the problem locally tool. I even used the same configuration.
I will rebase it and check again.

**Comment by PolynomialDivision** [comment]
Let's just rebase again. ;)

File: package/libs/wolfssl/Makefile [hunk 1]
```diff
@@ -8,12 +8,12 @@
...
 PKG_NAME:=wolfssl
-PKG_VERSION:=5.5.4-stable
-PKG_RELEASE:=$(AUTORELEASE)
+PKG_VERSION:=5.6.0-stable
+PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://github.com/wolfSSL/wolfssl/archive/v$(PKG_VERSION)
-PKG_HASH:=b7ee150e49def77c765bc02aac92ddeb0bebefd4cb12aa263d8f95e405221fb8
+PKG_HASH:=c1e689e21a17aa9b838e67a37c3eadfa578c8e260f8c77fb028c0316309f0636
```
Comment (DragonBluep) [inline package/libs/wolfssl/Makefile:16]: Package hash seems wrong, should be 979707380c677dfa65e3ba48f19e149773a4a32d ?
https://github.com/wolfSSL/wolfssl/releases/tag/v5.6.0-stable

**Comment by hauke** [comment]
I think this is a problem related to the ccache we use in the github actions. I am unable to reproduce it locally even when I also use ccache.
Maybe we should clean the ccache in the github action and then build this.

**Comment by hauke** [comment]
Thanks for disabling it, If that passes we can merge it and manually remove the ccache from github.

**Comment by kassane** [comment]
5.6.3: https://github.com/wolfSSL/wolfssl/commit/42e934b251dbd2c5a02a0f70eaba32c7f871b881

**Comment by PolynomialDivision** [comment]
> 5.6.3: [wolfSSL/wolfssl@42e934b](https://github.com/wolfSSL/wolfssl/commit/42e934b251dbd2c5a02a0f70eaba32c7f871b881)

What do you mean?

**Comment by kassane** [comment]
> > 5.6.3: [wolfSSL/wolfssl@42e934b](https://github.com/wolfSSL/wolfssl/commit/42e934b251dbd2c5a02a0f70eaba32c7f871b881)
> 
> What do you mean?

This is the new version's changelog (4 bugs fix).

**Comment by PolynomialDivision** [comment]
But there is not official release yet?
https://github.com/wolfSSL/wolfssl/releases

**Comment by kassane** [comment]
https://github.com/wolfSSL/wolfssl/releases/tag/v5.6.3-stable

**Comment by hauke** [comment]
I ran this in the build bot container and I got this error with wolfssl 5.6.2:
````
config.status: executing wolfssl/wolfcrypt/async.h commands
config.status: executing wolfssl/wolfcrypt/fips.h commands
config.status: executing wolfssl/wolfcrypt/port/cavium/cavium_nitrox.h commands
config.status: executing wolfssl/wolfcrypt/port/intel/quickassist.h commands
config.status: executing wolfssl/wolfcrypt/port/intel/quickassist_mem.h commands
configure: WARNING: unrecognized options: --disable-nls
configure: ---
configure: Running make clean...
configure: ---
configure: Generating user options header...
configure: error: Could not find colrm or cut to make options file
make[2]: *** [Makefile:230: /workspaces/openwrt/build_dir/target-mips_24kc_musl/wolfssl-regular/wolfssl-5.6.2-stable/.configured_dceadb4963ca5e127fd5b16e3762093b] Error 1
make[2]: Leaving directory '/workspaces/openwrt/package/libs/wolfssl'
time: package/libs/wolfssl/regular/compile#35.64#2.52#38.54
    ERROR: package/libs/wolfssl failed to build (build variant: regular).
make[1]: *** [package/Makefile:120: package/libs/wolfssl/compile] Error 1
make[1]: Leaving directory '/workspaces/openwrt'
make: *** [/workspaces/openwrt/include/toplevel.mk:232: package/wolfssl/compile] Error 2
buildbot@codespaces-a3fd99:/workspaces/openwrt$ 
````

Neither colrm nor cut are installed in the container.
The dependecy was added here: https://github.com/wolfSSL/wolfssl/commit/c3a56987990fea09404980b0f9d024ba4b14807f#diff-49473dca262eeab3b4a43002adb08b4db31020d190caaad1594b47f1d5daa810R8606

Later they added also checks if colrm is installed and then also used the alternative cut when it is not installed. 

We should probably add colrm or cut into our container, I am surprised it is not included. 

Debian packages providing colrm:
https://packages.debian.org/search?searchon=contents&keywords=colrm&mode=exactfilename&suite=oldstable&arch=any
Debian packages providing cut:
https://packages.debian.org/search?searchon=contents&keywords=cut&mode=exactfilename&suite=oldstable&arch=any

I would suggest to install coreutils.

Hmm, it looks like cut is installed:
````
buildbot@codespaces-a3fd99:/workspaces/openwrt$ colrm
bash: colrm: command not found
buildbot@codespaces-a3fd99:/workspaces/openwrt$ cut
cut: you must specify a list of bytes, characters, or fields
Try 'cut --help' for more information.
buildbot@codespaces-a3fd99:/workspaces/openwrt$ 
````

**Comment by hauke** [comment]
This patch is fixing the wolfssl build in the github actions: https://github.com/wolfSSL/wolfssl/commit/41d248461bd9ad44193a4806ecb5361513e8944e
Please backport it. 

Maybe we should upgrade to version 5.6.3 already, https://github.com/wolfSSL/wolfssl/releases/tag/v5.6.3-stable

**Comment by hauke** [comment]
You need this change to make the CI happy: https://github.com/wolfSSL/wolfssl/commit/41d248461bd9ad44193a4806ecb5361513e8944e

**Comment by PolynomialDivision** [comment]
Merged:
https://github.com/openwrt/openwrt/commit/0e83b5e6cc8e2970905a2b32c990fa7491ff733c

**Pull request closed without merging**